.eslintrc.js

```diff
@@ -21,11 +28,278 @@ module.exports = {
+			env: { es2021: true, node: true, 'jest/globals': true },
+
+			plugins: ['jest'],
+
+			rules: {
+				// enforce curly brace usage
+				curly: ['error', 'all'],
+
+				// allow class methods which do not use this
+				'class-methods-use-this': 'off',
+
+				// Allow use of ForOfStatement - no-restricted-syntax does not allow us to turn off a rule. This block overrides the airbnb rule entirely
+				// https://github.com/airbnb/javascript/blob/7152396219e290426a03e47837e53af6bcd36bbe/packages/eslint-config-airbnb-base/rules/style.js#L257-L263
+				'no-restricted-syntax': [
+					'error',
+					{
+						selector: 'ForInStatement',
+						message:
+							'for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array.',
+					},
+					{
+						selector: 'LabeledStatement',
+						message:
+							'Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.',
+					},
+					{
+						selector: 'WithStatement',
+						message:
+							'`with` is disallowed in strict mode because it makes code impossible to predict and optimize.',
+					},
+				],
+
+				// underscore dangle will be handled by @typescript-eslint/naming-convention
+				'no-underscore-dangle': 'off',
+
+				// enforce consistent sort order
+				'sort-imports': ['error', { ignoreCase: true, ignoreDeclarationSort: true }],
+
+				// disable import/no-cycle due to performance reasons; dpdm will validate this in CI
+				'import/no-cycle': 'off',
+
+				// enforce convention in import order
+				'import/order': [
+					'error',
+					{
+						'newlines-between': 'never',
+						groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
+						alphabetize: { order: 'asc', caseInsensitive: true },
+					},
+				],
+
+				// ensure consistent array typings
+				'@typescript-eslint/array-type': 'error',
+
+				// prefer type imports and exports
+				'@typescript-eslint/consistent-type-exports': [
+					'error',
+					{ fixMixedExportsWithInlineTypeSpecifier: true },
+				],
+				'@typescript-eslint/consistent-type-imports': ['error', { prefer: 'type-imports' }],
+
+				// enforce consistent order of class members
+				'@typescript-eslint/member-ordering': 'error',
+
+				// disallow parameter properties in favor of explicit class declarations
+				'@typescript-eslint/no-parameter-properties': 'error',
+
+				// ensure unused variables are treated as an error
+				// overrides @typescript-eslint/recommended -- '@typescript-eslint/no-unused-vars': 'warn'
+				// https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
+				'@typescript-eslint/no-unused-vars': 'error',
+			},
+		},
+		{
+			files: tsGlobs,
+			extends: ['plugin:@typescript-eslint/recommended-requiring-type-checking'],
+			rules: {
+				// ban ts-comment except with description
+				'@typescript-eslint/ban-ts-comment': [
```

Comment:
I think we should make this:
```
				// ban ts-comment except with description
				'@typescript-eslint/ban-ts-comment': [
					'error',
					{
						'ts-expect-error': { descriptionFormat: '^: ' },
						'ts-ignore': true,
						'ts-nocheck': true,
						'ts-check': false,
					},
				],
```

Reply:
Good idea. Looking at all these rules has been great. Thank you.